webserver/webserver/src/main/java/io/helidon/webserver/http/RoutingRequest.java

```diff
@@ -38,4 +38,14 @@ public interface RoutingRequest extends ServerRequest {
      * @return this instance
      */
     RoutingRequest prologue(HttpPrologue newPrologue);
+
+    /**
+     * Update the pattern used to match this request. Such as "/foo/{bar}".
+     *
+     * @param matchingPattern the matching patter
+     * @return this instance
+     */
+    default RoutingRequest matchingPattern(String matchingPattern) {
```

Comment:
I do not find it great that this method is visible to users when using `ServerRequest`. This method should never ever be called by users, so it should be hidden from them.

Reply:
Isn't that the case about all the methods in `RoutingRequest`?

Reply:
Added comment